src/include/parser/table_ref.h

```diff
@@ -49,17 +49,17 @@ class JoinDefinition {
   /**
    * @return left table
    */
-  std::shared_ptr<TableRef> GetLeftTable() { return left_; }
+  common::ManagedPointer<TableRef> GetLeftTable() { return common::ManagedPointer(left_); }
 
   /**
    * @return right table
    */
-  std::shared_ptr<TableRef> GetRightTable() { return right_; }
+  common::ManagedPointer<TableRef> GetRightTable() { return common::ManagedPointer(right_); }
 
   /**
    * @return join condition
    */
-  std::shared_ptr<AbstractExpression> GetJoinCondition() { return condition_; }
+  common::ManagedPointer<AbstractExpression> GetJoinCondition() { return common::ManagedPointer(condition_); }
```

Comment:
condition_ is already stored as a managed pointer is the wrapper needed?

Reply:
Nope, a remnant of worse days.